cypress/integration/articleFlows/mangageFlows/lockDiscussion.spec.js

```diff
@@ -0,0 +1,107 @@
+      cy.testSetup();
+      cy.fixture('users/articleEditorV1User.json').as('user');
+
+      cy.get('@user').then((user) => {
+        cy.loginUser(user).then(() => {
+          cy.createArticle({
+            title: 'Test Article',
+            tags: ['beginner', 'ruby', 'go'],
+            content: `This is a test article's contents.`,
+            published: true,
+          }).then((response) => {
+            cy.visit(`${response.body.current_state_path}/manage`);
+          });
+        });
+      });
+    });
+
+    it('should allow a user to lock a discussion', () => {
+      getDiscussionLockButton().click();
+      getDiscussionLockSubmitButton().click();
+
+      getDiscussionUnlockButton().should('exist');
+    });
+
+    it('should ask the user to confirm the discussion lock', () => {
+      getDiscussionLockButton().click();
+
+      cy.findByRole('heading', {
+        name: 'Are you sure you want to lock the discussion on this article?',
+      }).should('exist');
+
+      cy.findByRole('button', {
+        name: /Lock discussion/,
+      }).should('exist');
+    });
+
+    it('should allow a user to supply a reason and notes', () => {
+      getDiscussionLockButton().click();
+      cy.get('input[name="discussion_lock[notes]"]').should('exist');
+      cy.get('input[name="discussion_lock[reason]"]').should('exist');
+    });
+  });
+
+  describe('when an article has its discussion locked', () => {
+    beforeEach(() => {
+      cy.testSetup();
+      cy.fixture('users/articleEditorV1User.json').as('user');
+
+      cy.get('@user').then((user) => {
+        cy.loginUser(user).then(() => {
+          cy.createArticle({
+            title: 'Test Article',
+            tags: ['beginner', 'ruby', 'go'],
+            content: `This is a test article's contents.`,
+            published: true,
+          }).then((response) => {
+            const articlePath = response.body.current_state_path;
+            cy.visit(`${articlePath}/manage`);
+            getDiscussionLockButton().click();
+            cy.get('#discussion_lock_reason')
+              .should('be.visible')
+              .type(exampleReason);
+            cy.get('#discussion_lock_notes')
+              .should('be.visible')
+              .type(exampleNotes);
+            getDiscussionLockSubmitButton().click();
+            cy.visit(articlePath);
+          });
+        });
+      });
+    });
+
+    it('should show the discussion lock on an article', () => {
+      cy.get('#discussion-lock').should('exist');
+    });
+
+    it('should show the discussion lock reason on an article', () => {
+      cy.get('#discussion-lock').should(($discussionLock) => {
+        expect($discussionLock).to.contain.text(exampleReason);
+      });
+    });
+
+    it('should not show the discussion lock notes on an article', () => {
+      cy.get('#discussion-lock').should(($discussionLock) => {
+        expect($discussionLock).not.to.contain.text(exampleNotes);
+      });
+    });
```

Comment:
Just a thought - could these verification steps be tacked on the end of the previous tests e.g. "should allow a user to lock a discussion"? I'm just thinking when you lock the discussion, you're on the "manage" page and could then just click the article link and check the notice appears?

It would save the additional test setup for these ones, and it would mean the tests cover the full flow from being able to submit the form to seeing the correct UI on the article page.

No worries if there's a reason that doesn't work though.

Reply:
I didn't do that for 2 reasons:
1. Managing the state to get the article path is annoying. The before actions here are different than the ones above.
2. The _displaying_ of discussion locks and the _creation/deletion_, in my mind, are 2 totally different things. The code for each doesn't touch the other. I figured this "degree of separation" in the test would also reflect that. Technically, we could create discussion locks and temporarily disable displaying them for some reason. In that edge case, these tests would be "easier" to manage - skip the second set.

If we really feel strongly about it, we can smash it all into one.

Reply:
I don't feel strongly about it at all 😄  That all makes sense!

I was thinking more from the user POV, after you lock a discussion you would naturally want to see some change reflected on the article page so it would be one user flow (and by the time you get to the "Manage" page you have a link to the article you can click, so we don't need to worry about the article path state any more). 

But I'm happy for them to be separate, it was just a thought!